compiler/src/main/kotlin/ast/ast.kt

```diff
@@ -9,21 +9,52 @@ import org.antlr.v4.runtime.tree.ParseTree
  */
 const val MAGIC_UNDEFINED_STRING = "<<THIS IS A MAGIC STRING, UNDEFINED>>"
 
+/**
+ * This classes is the basis for all nodes in the abstract syntax tree.
+ *
+ * It's sealed such that it's possible to use Kotlin's when statement to without an else.
```

Comment:
I think an extra "to" made it's way in
```suggestion
 * It's sealed such that it's possible to use Kotlin's when statement without an else.
```